### DISCUSSION TITLE
Get type on fetcher.data

### DISCUSSION THREAD

**Discussion by treasurely** [remix-run/remix#4977] [Q&A] [answered]
Does anyone know if theres' a way/best practice on getting the type of a fetcher's `.data` after it calls a loader that's located in a resource route(similar to a `useLoaderData<typeof loader>()` in a normal route)?  I see references to a `useFetcher<typeof loader>()` syntax in some of the docs but nothing concrete on how it works.

Thanks!

**Comment by sergiodxa** [comment, marked as answer]
Pass typeof loader to the useFetcher hook and it will type fetcher.data as what the loader (or action if typeof action) returns

**Reply by treasurely** [reply]
Ah! I had to import the loader with `import { loader } from "~/routes/<<resource_route>>`, now I have it working.  Thanks!
